scripts/translate_java_test_strings.py

```
@@ -1,89 +1,304 @@
+                )
+
+                # Replace the region [remove_start:insert_pos] with new_code
+                source_text = source_text[:remove_start] + new_code + source_text[insert_pos:]
+                updated = True
+
+            else:
+                # No else branch: insert after current.end_byte. Ensure proper newline separation.
+                insert_pos = current.end_byte
+                # Find if we are mid-line: check previous newline
+                prev_newline = source_text.rfind('\n', 0, insert_pos)
+                prev_char = source_text[insert_pos - 1] if insert_pos > 0 else '\n'
```

Ruff (0.14.10)

Local variable `prev_char` is assigned to but never used

Remove assignment to unused variable `prev_char`

(F841)